packages/i18n/src/index.js

```diff
@@ -1,63 +1,51 @@
 /**
  * External dependencies
  */
-import Jed from 'jed';
-import memoize from 'memize';
+import Tannin from 'tannin';
 
-let i18n;
+export { default as sprintf } from '@tannin/sprintf';
 
 /**
- * Log to console, once per message; or more precisely, per referentially equal
- * argument set. Because Jed throws errors, we log these to the console instead
- * to avoid crashing the application.
+ * Default locale data to use for Tannin domain when not otherwise provided.
+ * Assumes an English plural forms expression.
  *
- * @param {...*} args Arguments to pass to `console.error`
+ * @type {Object}
  */
-const logErrorOnce = memoize( console.error ); // eslint-disable-line no-console
+const DEFAULT_LOCALE_DATA = {
+	'': {
+		plural_forms: 'plural=(n!=1)',
+	},
+};
+
+const i18n = new Tannin( {} );
 
 /**
- * Merges locale data into the Jed instance by domain. Creates a new Jed
- * instance if one has not yet been assigned.
+ * Merges locale data into the Tannin instance by domain. Accepts data in a
+ * Jed-formatted JSON object shape.
  *
  * @see http://messageformat.github.io/Jed/
  *
- * @param {?Object} localeData Locale data configuration.
- * @param {?string} domain     Domain for which configuration applies.
+ * @param {?Object} data   Locale data configuration.
+ * @param {?string} domain Domain for which configuration applies.
  */
-export function setLocaleData( localeData = { '': {} }, domain = 'default' ) {
-	if ( ! i18n ) {
-		i18n = new Jed( {
-			domain: 'default',
-			locale_data: {
-				default: { '': {} },
-			},
-		} );
-	}
+export function setLocaleData( data, domain = 'default' ) {
+	i18n.data[ domain ] = {
+		...DEFAULT_LOCALE_DATA,
+		...i18n.data[ domain ],
+		...data,
+	};
 
-	i18n.options.locale_data[ domain ] = Object.assign(
-		{},
-		i18n.options.locale_data[ domain ],
-		localeData
-	);
+	// Populate default domain configuration (supported locale date which omits
+	// a plural forms expression).
+	i18n.data[ domain ][ '' ] = {
+		...DEFAULT_LOCALE_DATA[ '' ],
+		...i18n.data[ domain ][ '' ],
+	};
 }
 
 /**
- * Returns the current Jed instance, initializing with a default configuration
- * if not already assigned.
- *
- * @return {Jed} Jed instance.
- */
-function getI18n() {
-	if ( ! i18n ) {
-		setLocaleData();
-	}
-
-	return i18n;
-}
```

Comment:
Do we need a replacement for this?

Reply:
No, I don't think so. It's not exposed as a public API, and its previous purpose of in-time initialization is now served instead by combination of initialize-at-assignment and preparing domain data within `dcnpgettext`.